app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/vm.rb

```diff
@@ -104,6 +107,15 @@ def console_supported?(type)
     return true if type.upcase == 'VNC'
   end
 
+  def console_url
+    crn = CGI.escape(ext_management_system.pcloud_crn.values.join(":"))
+    "https://cloud.ibm.com/services/power-iaas/#{crn}/server/#{uid_ems}/?paneId=manage&crn=#{crn}"
```

Comment:
Actually, I just realized you are using this for building the query part of the URL...in that case, I think you should use `URI.encode_www_form`:

```suggestion
    params = URI.encode_www_form(:paneId => "manageiq", :crn => ext_management_system.pcloud_crn.values.join(":"))
    "https://cloud.ibm.com/services/power-iaas/#{crn}/server/#{uid_ems}/?params"
```

Reply:
Feedback incorporated, force-push'd. PTAL.